### DISCUSSION TITLE
DisplayObject regX, regY percentage

### DISCUSSION THREAD

**Discussion by ghost** [airsdk/Adobe-Runtime-Support#3376] [Ideas]
![image](https://github.com/user-attachments/assets/008285fb-f23e-424c-aaee-2946c7af732f)

**Comment by ajwfrost** [comment]
Sounds quite a nice and straightforward idea.

This is kind of possible already (and happens in Animate) by an object having bounds that go from negative to positive e.g. from (-100, -20) to (100, 20) for an object that's 200x40 pixels. So - what sort of DisplayObject are you thinking of here, where you would only have it with contents in the positive direction? Text fields are one I guess...

As an aside, you could write a wrapper class based on DisplayObjectContainer to do this automatically I think. Which might be a first step towards then building the API into AIR..

thanks

**Reply by ghost** [reply]
> So - what sort of DisplayObject are you thinking of here, where you would only have it with contents in the positive direction?

It is usually the rectangles comprising a player such as a 2D top-down automobile.

> As an aside, you could write a wrapper class based on DisplayObjectContainer to do this automatically I think. Which might be a first step towards then building the API into AIR..

Good idea; a wrapper that listens to enterFrame might work

I think `RotationRegObject`, similiar to [that center-axis rotation formula](https://github.com/agera-air/util/blob/master/src/agera/util/DisplayListOrientation.as) I found in a community.

**Reply by marchbold** [reply]
I used a lib with similar functionalities many years ago, so it's definitely possible to just do in a wrapper. Wish I could remember the lib for you.

**Reply by itlancer** [reply]
We also use wrappers for that. But will be glad to have it as a part of `DisplayObject` APIs.

**Reply by itlancer** [reply]
Reference from CSS - `transform-origin`: https://developer.mozilla.org/en-US/docs/Web/CSS/transform-origin

**Comment by henke37** [comment]
If you are unhappy about the transformation point, use the flash.geom.Matrix class to compute your own transform matrix with whatever transform point you like.

**Comment by pol2095** [comment]
Starling use properties [pivotX](https://doc.starling-framework.org/current/starling/display/DisplayObject.html#pivotX), [pivotY](https://doc.starling-framework.org/current/starling/display/DisplayObject.html#pivotY) and [pivotZ](https://doc.starling-framework.org/current/starling/display/Sprite3D.html#pivotZ), is-it the same ?

**Reply by henke37** [reply]
Probably.
